### PR TITLE
NEUBURG-101: Use refPath for dealing with multiple forms

### PR DESCRIPTION
**Pull request by NKlug**

### CODE REVIEW

**Review by michael-markl — COMMENTED** [review]
Comment

File: src/models/Offer.js [hunk 1]
```diff
@@ -35,7 +34,7 @@ const offerSchema = mongoose.Schema({
   },
   formData: {
     type: mongoose.Schema.Types.ObjectId,
-    ref: Array.from(Object.keys(forms)), // Must be one of the forms defined in forms.js
```
Comment (michael-markl) [inline src/models/Offer.js:38]: So this creates collections with the name 'neub[...]wohnraum' or similar, right? 
Do we somehow assure, that we can only PUT offers with an existing city (and according formData scheme) ?

Comment (NKlug) [reply src/models/Offer.js:38]: Yeah, you're right. The city name is used for finding the right form for population (and therefore must always match the model's name in mongoose).
Right now we already assure that we can't enter forms without existing cities, check [this](https://github.com/Integreat/neuburg-backend/blob/4f9de7bb69acc2e74029ffd632a74e7664ff50f3/src/services/OfferService.js#L33) for reference. The `FormModel` will always be `undefined` here, which gives an error. Btw this error is not handled yet, I created a task for it.

File: src/services/OfferService.js [hunk 2]
```diff
@@ -84,6 +84,9 @@ export default class OfferService {
       .lean()
       .exec()
 
+    // Can't exclude the city in the projection because of populate
+    offers.forEach(offer => delete offer.city)
```
Comment (michael-markl) [inline src/services/OfferService.js:88]: Can you do another select after the populate? (Just curious^^)

File: src/models/Offer.js (hunk 1 above)
Comment (michael-markl) [reply src/models/Offer.js:38]: Do we need to explicitly define the model, or would mongoose create an empty one, if it doesn't find one?

Do we need to migrate the database? (assuming there is some data in it^^)

**Review by michael-markl — APPROVED** [review]

File: src/services/OfferService.js (hunk 2 above)
Comment (NKlug) [reply src/services/OfferService.js:88]: Tried it, but didn't work (for me). I think the thing is that mongoose is just submitting the query to the mongodb and mongodb deals with the exact order of execution.

File: src/models/Offer.js (hunk 1 above)
Comment (NKlug) [reply src/models/Offer.js:38]: Although I'm not sure what mongoose would do (probably throw an error I guess) this situation does not really occur. We are first creating and saving the form for each city. That's the point we would have to handle non existing cities. Only after that we create the actual `Offer` with a link to the form.

I don't think DB migration is necessary, but I'm not sure. Afaik there are no offers yet, so it shouldn't cause any problem :D

Comment (michael-markl) [reply src/models/Offer.js:38]: Alright, we should still be careful and double check when deploying this^^...

**Pull request merged**